apps/src/appsApi.js

```diff
@@ -45,31 +56,53 @@ var ApiRequestHelper = function (baseUrl) {
   this.apiBaseUrl_ = baseUrl;
 };
 
+/**
+ * @param {!string} localUrl - API endpoint relative to API base URL.
+ * @param {!NodeStyleCallback} callback
+ * @param {*} failureValue - what to pass as the callback result if an error
+ *        occurs.
+ */
 ApiRequestHelper.prototype.get = function (localUrl, callback, failureValue) {
   $.ajax({
     url: this.apiBaseUrl_ + localUrl,
     type: 'get',
     dataType: 'json'
-  }).done(function (result /*, text*/) {
-    callback(result);
-  }).fail(function (/*request, status, error*/) {
-    callback(failureValue);
+  }).done(function (data /*, textStatus, jqXHR*/) {
+    callback(null, data);
+  }).fail(function (jqXHR, textStatus, errorThrown) {
+    callback(
+        new Error('textStatus: ' + textStatus + '; errorThrown: ' + errorThrown),
+        failureValue);
```

Comment:
I'm not sure we need failureValue here. The way I've usually seen this done is that if we have an error, the result is either null or garbage (i.e. we cant depend on it being anything in particular). Instead we can have a callback that looks something like this:
```
foo.get(myLocalUrl, function (err, result) {
  if (err !== null) {
    // handle failure. if we need some sort of failure value it can live here
    return;
  }
  // handle success
}